src/modules/musig/main_impl.h

```diff
@@ -45,23 +45,36 @@ static void secp256k1_musig_sha256_init_tagged(secp256k1_sha256 *sha) {
     sha->bytes = 64;
 }
 
-/* Compute r = SHA256(ell, x). Assumes field element x is normalized. */
-static void secp256k1_musig_coefficient(secp256k1_scalar *r, const unsigned char *ell, secp256k1_fe *x) {
+/* Compute MuSig coefficient which is constant 1 for the second pubkey and
+ * SHA256(ell, x) otherwise. second_pk_x can be NULL in case there is no
+ * second_pk. Assumes both field elements x and second_pk_x are normalized. */
+static void secp256k1_musig_coefficient_internal(secp256k1_scalar *r, const unsigned char *ell, secp256k1_fe *x, const secp256k1_fe *second_pk_x) {
     secp256k1_sha256 sha;
     unsigned char buf[32];
 
-    secp256k1_musig_sha256_init_tagged(&sha);
-    secp256k1_sha256_write(&sha, ell, 32);
-    secp256k1_fe_get_b32(buf, x);
-    secp256k1_sha256_write(&sha, buf, 32);
-    secp256k1_sha256_finalize(&sha, buf);
-    secp256k1_scalar_set_b32(r, buf, NULL);
+    if (secp256k1_fe_cmp_var(x, second_pk_x) == 0) {
+        secp256k1_scalar_set_int(r, 1);
+    } else {
+        secp256k1_musig_sha256_init_tagged(&sha);
+        secp256k1_sha256_write(&sha, ell, 32);
+        secp256k1_fe_get_b32(buf, x);
+        secp256k1_sha256_write(&sha, buf, 32);
+        secp256k1_sha256_finalize(&sha, buf);
+        secp256k1_scalar_set_b32(r, buf, NULL);
+    }
+}
+
+static void secp256k1_musig_coefficient(secp256k1_scalar *r, const secp256k1_musig_pre_session *pre_session, secp256k1_fe *x) {
```

Comment:
Also add a comment here like "Assumes both field elements x and second_pk_x are normalized."? Or move the existing one here.

Reply:
done